pkg/resources/network_policy.go

```diff
@@ -151,149 +181,168 @@ func ReadContextNetworkPolicy(ctx context.Context, d *schema.ResourceData, meta
 	}
 
 	if err = d.Set("comment", networkPolicy.Comment); err != nil {
 		return diag.FromErr(err)
 	}
 
-	policyDescriptions, err := client.NetworkPolicies.Describe(ctx, id)
+	policyProperties, err := client.NetworkPolicies.Describe(ctx, id)
 	if err != nil {
 		return diag.FromErr(err)
 	}
-	for _, desc := range policyDescriptions {
-		switch desc.Name {
-		case "ALLOWED_IP_LIST":
-			if err = d.Set("allowed_ip_list", strings.Split(desc.Value, ",")); err != nil {
-				return diag.FromErr(err)
-			}
-		case "BLOCKED_IP_LIST":
-			if err = d.Set("blocked_ip_list", strings.Split(desc.Value, ",")); err != nil {
-				return diag.FromErr(err)
-			}
-		case "ALLOWED_NETWORK_RULE_LIST":
-			var networkRules []NetworkRulesSnowflakeDTO
-			err := json.Unmarshal([]byte(desc.Value), &networkRules)
-			if err != nil {
-				return diag.FromErr(err)
-			}
-			networkRulesFullyQualified := make([]string, len(networkRules))
-			for i, ele := range networkRules {
-				networkRulesFullyQualified[i] = sdk.NewSchemaObjectIdentifierFromFullyQualifiedName(ele.FullyQualifiedRuleName).FullyQualifiedName()
-			}
 
-			if err = d.Set("allowed_network_rule_list", networkRulesFullyQualified); err != nil {
-				return diag.FromErr(err)
-			}
-		case "BLOCKED_NETWORK_RULE_LIST":
-			var networkRules []NetworkRulesSnowflakeDTO
-			err := json.Unmarshal([]byte(desc.Value), &networkRules)
-			if err != nil {
-				return diag.FromErr(err)
-			}
-			networkRulesFullyQualified := make([]string, len(networkRules))
-			for i, ele := range networkRules {
-				networkRulesFullyQualified[i] = sdk.NewSchemaObjectIdentifierFromFullyQualifiedName(ele.FullyQualifiedRuleName).FullyQualifiedName()
-			}
+	log.Println("PROPS:", policyProperties)
 
-			if err = d.Set("blocked_network_rule_list", networkRulesFullyQualified); err != nil {
-				return diag.FromErr(err)
-			}
+	if allowedIpList, err := collections.FindOne(policyProperties, func(prop sdk.NetworkPolicyProperty) bool { return prop.Name == "ALLOWED_IP_LIST" }); err == nil {
+		if err = d.Set("allowed_ip_list", sdk.ParseCommaSeparatedStringArray(allowedIpList.Value, false)); err != nil {
+			return diag.FromErr(err)
+		}
+	} else {
+		if err = d.Set("allowed_ip_list", []any{}); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	if blockedIpList, err := collections.FindOne(policyProperties, func(prop sdk.NetworkPolicyProperty) bool { return prop.Name == "BLOCKED_IP_LIST" }); err == nil {
+		if err = d.Set("blocked_ip_list", sdk.ParseCommaSeparatedStringArray(blockedIpList.Value, false)); err != nil {
+			return diag.FromErr(err)
+		}
+	} else {
+		if err = d.Set("blocked_ip_list", []any{}); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	allowedNetworkRules := make([]string, 0)
+	if allowedNetworkRuleList, err := collections.FindOne(policyProperties, func(prop sdk.NetworkPolicyProperty) bool { return prop.Name == "ALLOWED_NETWORK_RULE_LIST" }); err == nil {
+		var networkRules []sdk.NetworkRulesSnowflakeDTO
```

Comment:
Can this be extracted? It's the same as below.

Reply:
Sure